UpdateHealthInfoCacheActionTests testRequestSerialization failing

### CI Link

https://gradle-enterprise.elastic.co/s/veux6bidkudnc

### Repro line

./gradlew ':server:test' --tests "org.elasticsearch.health.node.UpdateHealthInfoCacheActionTests.testRequestSerialization" -Dtests.seed=1B12EE4C2D698950 -Dtests.locale=sr-Latn-RS -Dtests.timezone=Atlantic/St_Helena -Druntime.java=21

### Does it reproduce?

Yes

### Applicable branches

main

### Failure history

6 failures on feb 7, good before that: https://gradle-enterprise.elastic.co/scans/tests?search.timeZoneId=America%2FVancouver&tests.container=org.elasticsearch.health.node.UpdateHealthInfoCacheActionTests&tests.test=testRequestSerialization

### Failure excerpt

```
> Task :server:test
Feb 07, 2024 9:35:09 AM sun.util.locale.provider.LocaleProviderAdapter <clinit>
WARNING: COMPAT locale provider will be removed in a future release

REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.health.node.UpdateHealthInfoCacheActionTests.testRequestSerialization" -Dtests.seed=1B12EE4C2D698950 -Dtests.locale=sr-Latn-RS -Dtests.timezone=Atlantic/St_Helena -Druntime.java=21

org.elasticsearch.health.node.UpdateHealthInfoCacheActionTests > testRequestSerialization FAILED
    java.lang.AssertionError: Request mutation should not be equal to original
    Expected: not <org.elasticsearch.health.node.UpdateHealthInfoCacheAction$Request/unset>
         but: was <org.elasticsearch.health.node.UpdateHealthInfoCacheAction$Request/unset>
        at __randomizedtesting.SeedInfo.seed([1B12EE4C2D698950:DA5E25B292C51AD5]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:964)
        at org.elasticsearch.test.EqualsHashCodeTestUtils.checkEqualsAndHashCode(EqualsHashCodeTestUtils.java:95)
        at org.elasticsearch.test.EqualsHashCodeTestUtils.checkEqualsAndHashCode(EqualsHashCodeTestUtils.java:63)
        at org.elasticsearch.health.node.UpdateHealthInfoCacheActionTests.testRequestSerialization(UpdateHealthInfoCacheActionTests.java:117)

WARNING: A terminally deprecated method in java.lang.System has been called
WARNING: System::setSecurityManager has been called by org.gradle.api.internal.tasks.testing.worker.TestWorker (file:/Users/john.verwolf/.gradle/wrapper/dists/gradle-8.5-all/3zlzzgtsutfj0pbojr50n2l7z/gradle-8.5/lib/plugins/gradle-testing-base-8.5.jar)
WARNING: Please consider reporting this to the maintainers of org.gradle.api.internal.tasks.testing.worker.TestWorker
WARNING: System::setSecurityManager will be removed in a future release

> Task :server:test FAILED


Suite: Test class org.elasticsearch.health.node.UpdateHealthInfoCacheActionTests
  1> [2024-02-07T17:35:15,985][INFO ][o.e.h.n.UpdateHealthInfoCacheActionTests] [testRequestSerialization] before test
  1> [2024-02-07T17:35:16,262][INFO ][o.e.h.n.UpdateHealthInfoCacheActionTests] [testRequestSerialization] after test
  2> REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.health.node.UpdateHealthInfoCacheActionTests.testRequestSerialization" -Dtests.seed=1B12EE4C2D698950 -Dtests.locale=sr-Latn-RS -Dtests.timezone=Atlantic/St_Helena -Druntime.java=21
  2> java.lang.AssertionError: Request mutation should not be equal to original
    Expected: not <org.elasticsearch.health.node.UpdateHealthInfoCacheAction$Request/unset>
         but: was <org.elasticsearch.health.node.UpdateHealthInfoCacheAction$Request/unset>
        at __randomizedtesting.SeedInfo.seed([1B12EE4C2D698950:DA5E25B292C51AD5]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:964)
        at org.elasticsearch.test.EqualsHashCodeTestUtils.checkEqualsAndHashCode(EqualsHashCodeTestUtils.java:95)
        at org.elasticsearch.test.EqualsHashCodeTestUtils.checkEqualsAndHashCode(EqualsHashCodeTestUtils.java:63)
        at org.elasticsearch.health.node.UpdateHealthInfoCacheActionTests.testRequestSerialization(UpdateHealthInfoCacheActionTests.java:117)
  2> NOTE: leaving temporary files on disk at: /Users/john.verwolf/code/elasticsearch/server/build/testrun/test/temp/org.elasticsearch.health.node.UpdateHealthInfoCacheActionTests_1B12EE4C2D698950-002
  2> NOTE: test params are: codec=Asserting(Lucene99): {}, docValues:{}, maxPointsInLeafNode=1994, maxMBSortInHeap=6.7334910437814, sim=Asserting(RandomSimilarity(queryNorm=true): {}), locale=sr-Latn-RS, timezone=Atlantic/St_Helena
  2> NOTE: Mac OS X 14.3 aarch64/Oracle Corporation 21 (64-bit)/cpus=10,threads=1,free=455253952,total=536870912
  2> NOTE: All tests run in this JVM: [UpdateHealthInfoCacheActionTests]

Tests with failures:
 - org.elasticsearch.health.node.UpdateHealthInfoCacheActionTests.testRequestSerialization

1 test completed, 1 failed
```

--- a/server/src/main/java/org/elasticsearch/health/node/UpdateHealthInfoCacheAction.java
+++ b/server/src/main/java/org/elasticsearch/health/node/UpdateHealthInfoCacheAction.java
@@ -46,11 +46,16 @@ public class UpdateHealthInfoCacheAction extends ActionType<AcknowledgedResponse
         @Nullable
         private final RepositoriesHealthInfo repositoriesHealthInfo;
 
-        public Request(String nodeId, DiskHealthInfo diskHealthInfo, RepositoriesHealthInfo repositoriesHealthInfo) {
+        public Request(
+            String nodeId,
+            DiskHealthInfo diskHealthInfo,
+            DataStreamLifecycleHealthInfo dslHealthInfo,
+            RepositoriesHealthInfo repositoriesHealthInfo
+        ) {
             this.nodeId = nodeId;
             this.diskHealthInfo = diskHealthInfo;
+            this.dslHealthInfo = dslHealthInfo;
             this.repositoriesHealthInfo = repositoriesHealthInfo;
-            this.dslHealthInfo = null;
         }
 
         public Request(String nodeId, DataStreamLifecycleHealthInfo dslHealthInfo) {
@@ -179,7 +184,7 @@ public class UpdateHealthInfoCacheAction extends ActionType<AcknowledgedResponse
             }
 
             public Request build() {
-                return new Request(nodeId, diskHealthInfo, repositoriesHealthInfo);
+                return new Request(nodeId, diskHealthInfo, dslHealthInfo, repositoriesHealthInfo);
             }
         }
     }

